keras_cv/models/segmentation/deeplab_v3/deeplab_v3.py

```diff
@@ -0,0 +1,260 @@
+            relevant if `low_level_feature_layer_name` is also specified.
+        segmentation_head: (Optional) a `keras.layers.Layer`. If provided, the
+            outputs of the DeepLabV3 encoder is passed to this layer and it should
+            predict the segmentation mask based on feature from backbone and feature
+            from decoder, otherwise a similar architecture is used.
+        segmentation_head_activation: Optional `str` or function, activation
+            functions between the `keras.layers.Conv2D` layers and the final
+            classification layer, defaults to `"relu"`.
+        input_shape: optional shape tuple, defaults to `(None, None, 3)`.
+        input_tensor: optional Keras tensor (i.e., output of `layers.Input()`)
+            to use as image input for the model.
+
+    Examples:
+    ```python
+    import tensorflow as tf
+    import keras_cv
+
+    images = tf.ones(shape=(1, 96, 96, 3))
+    labels = tf.zeros(shape=(1, 96, 96, 1))
+    backbone = keras_cv.models.ResNet50V2Backbone(input_shape=[96, 96, 3])
+    model = keras_cv.model.segmentation.DeepLabV3(
+        num_classes=1,
+        backbone=backbone,
+        low_level_feature_layer_name="v2_stack_0_block1_2_relu"
+    )
+
+    # Evaluate model
+    model(images)
+
+    # Train model
+    model.compile(
+        optimizer="adam",
+        loss=keras.losses.BinaryCrossentropy(from_logits=False),
+        metrics=["accuracy"],
+    )
+    model.fit(images, labels, epochs=3)
+    ```
+    """
+
+    def __init__(
+        self,
+        num_classes,
+        backbone,
+        spatial_pyramid_pooling=None,
+        low_level_feature_layer_name=None,
+        projection_filters=None,
+        segmentation_head=None,
+        segmentation_head_activation="softmax",
+        dropout=0.0,
+        input_shape=(None, None, 3),
+        input_tensor=None,
+        **kwargs,
+    ):
+        if not isinstance(backbone, keras.layers.Layer) or not isinstance(
+            backbone, keras.Model
+        ):
+            raise ValueError(
+                "Argument `backbone` must be a `keras.layers.Layer` instance "
+                f" or `keras.Model`. Received instead "
+                f"backbone={backbone} (of type {type(backbone)})."
+            )
+
+        inputs = utils.parse_model_inputs(input_shape, input_tensor)
+
+        if input_shape[0] is None and input_shape[1] is None:
+            input_shape = backbone.input_shape[1:]
+            inputs = keras.Input(tensor=input_tensor, shape=input_shape)
+
+        if input_shape[0] is None and input_shape[1] is None:
+            raise ValueError(
+                "Input shapes for both the backbone and DeepLabV3 cannot be "
+                "`None`. Received: input_shape={input_shape} and "
+                "backbone.input_shape={backbone.input_shape[1:]}"
+            )
+
+        height = input_shape[0]
+        width = input_shape[1]
+
+        feature_map = backbone(inputs)
```

Comment:
Calling `backbone(inputs)` doesn't get us a full feature map -- it just gets the final feature set.

I _think_ this is intentional with DeepLab? (I'm not super familiar with the model), so we should just rename this to reflect that it's not a full feature map.

Reply:
@ianstenbit Do you think we should we specify the name of the layer for getting the feature maps for the encoder as well?

Reply:
Based on other code, it seems like this is expected to be the P4 output of the backbone. See my comment below -- basically, assuming I'm understanding the model correct, yes we should be extracting this from a specific layer -- the P4 output of the backbone.